### ISSUE TITLE
Fragment links are rendered different from regular links (missing underline)

### ISSUE THREAD

**Issue by Josh-Cena** [mdn/fred#716]
There are two things that make our anchor link handling non-ideal:

- They don't get the underline. This is anti-a11y: we used to not have the underline at all, and they were added because people with color blindness cannot perceive the color difference and need another visual cue. This argument obviously applies to anchor links as well, which are still links. Removing underlines means that these anchor links appear just like normal content to them.
- Manual anchor links in DTs, like `- [xxx](#yyy)`, don't get link styling. While I get the rationale for not styling autogenerated links, I believe that if a writer writes such a link manually, they would expect it to be indicated as a link. This could probably be fixed by injecting a class name in Rari when autogenerating anchor links to the DT itself, and then targeting that class name instead of targeting `a[href^="#"]`.

For example, on this page: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension

![image](https://github.com/user-attachments/assets/5fc7cfd0-9e82-4cab-8ba1-1fac7fdbf414)

Each DT is actually an explicit anchor link and it should take you somewhere else. (For some mysterious reason this doesn't seem to be all true, so perhaps there's always a Rari bug here.) But there's no styling to indicate this.

Furthermore consider _any_ anchor link:

![image](https://github.com/user-attachments/assets/8b3290ef-f986-4a53-8899-a9c9a59672c4)

If I remove the colors, then the fact that the second link is clickable is very inobvious. These two links are both links and there shouldn't be a distinction!

**Comment by caugner** [comment]
@Josh-Cena Thanks for opening this issue, can you please add screenshots to the issue description?

**Comment by caugner** [comment]
Note: This overlaps with https://github.com/mdn/fred/issues/702, which will be fixed with https://github.com/mdn/fred/pull/703.

**Comment by Josh-Cena** [comment]
While related, this issue stems from #703, so even with that PR merged, problems in this issue remain.

**Comment by globau** [comment]
@mdn/content-team

**Comment by caugner** [comment]
> While related, this issue stems from [#703](https://github.com/mdn/fred/pull/703), so even with that PR merged, problems in this issue remain.

To clarify:

- #703 (which landed on prod last night) restored the link _color_ of fragment links. This was an obvious bug.
- Fragment links continue to have inverted underline behavior compared to regular links. This appears to have been implemented intentionally.

I can see three options:

1. Keep as is (fragment links with inverted underline behavior)
2. Style fragment links exactly like regular links (users will no longer be able to distinguish them, apart from looking at the browser status bar)
3. Replace underline behavior different with other indicator (e.g. underline style, or ::after pseudo)

I personally think 2. is bad UX in the example above, unless content is changed to provide a semantical hint that this is a fragment link, e.g. by putting the link in brackets, like "(see below/above)".

**Comment by Josh-Cena** [comment]
Why should fragment links be different? They are still back/forward navigable and they take readers to a different context. Was there a discussion where someone said they need to distinguish between the two?

**Comment by globau** [comment]
FWIW I think option 2 - treating all links as equals - is the best route; as @Josh-Cena points out it's the change of context that's more important than the technical underpinnings.

**Comment by Rumyra** [comment]
@Josh-Cena just a quick update - we have just discussed this internally and I'll put together some options for sorting out link styling based on the issue

**Comment by pepelsbey** [comment]
Demos exploring possible solutions
 
- [Anchor links for titles](https://codepen.io/pepelsbey/pen/jErpzdN)
- [Anchor links for titles v1](https://codepen.io/pepelsbey/pen/jErpzpj)
- [Anchor links for titles v2](https://codepen.io/pepelsbey/pen/vEKpeJz)

**Comment by pepelsbey** [comment]
[The troublesome bit](https://github.com/mdn/fred/blob/d99b5dbec5721c96435fe1aabff1305f4b728adb/components/content-section/server.css#L59-L67)

```css
[href^="#"]:is(.content-section a) {
  text-decoration: none;
}
```

**Comment by Rumyra** [comment]
> Furthermore consider any anchor link

It is this last part of the original issue which is unresolved - the decision is to underline all links and not differentiate between links that go to a different page and links that go to the same page - @pepelsbey has identified the code to modify for this above

**Comment by globau** [comment]
@pepelsbey do you want to submit a PR for this?
